azure/azure-release-template.yml

```diff
@@ -0,0 +1,128 @@
+parameters:
+  - name: service_name
+    type: string
+  - name: short_service_name
+    type: string
+  - name: service_base_path
+    type: string
+  - name: product_display_name
+    type: string
+  - name: product_description
+    type: string
+  - name: service_base_path_pr
+    type: string
+  - name: spec_file
+    type: string
+  - name: apigee_deployments
+    type: object
+
+extends:
+  template: azure/common/apigee-deployment.yml@common
+  parameters:
+    service_name: ${{ parameters.service_name }}
+    short_service_name: ${{ parameters.short_service_name }}
+    service_base_path: ${{ parameters.service_base_path }}
+    product_display_name: ${{ parameters.product_display_name }}
+    product_description: ${{ parameters.product_description }}
+    spec_file: ${{ parameters.spec_file }}
+    apigee_deployments: ${{ parameters.apigee_deployments }}
+    post_deploy:
+      - task: DockerInstaller@0
+        displayName: Install docker
+        inputs:
+          dockerVersion: 17.09.0-ce
+          releaseType: stable
+      - bash: |
+          # Set variables
+          export SERVICE_NAME=$(SERVICE_NAME)
+          export SERVICE_ARTIFACT_NAME=`ls $SERVICE_NAME`
+          
+          if [ ! -d "$SERVICE_NAME/$SERVICE_ARTIFACT_NAME/pact" ]; then
+              echo "No pacts found. Exiting"
+            exit 0
+          fi
+          
+          if [[ $SERVICE_ARTIFACT_NAME == v* ]]; then
+            export BUILD_VERSION=`echo $SERVICE_ARTIFACT_NAME | grep -o "v[0-9]\+\.[0-9]\+\.[0-9]\+-[[:alpha:]]\+" | tail -1`
+            export SERVICE_BASE_PATH="${{ parameters.service_base_path }}"
+          else
+            export BUILD_VERSION="${{ parameters.service_base_path_pr }}"
+            export SERVICE_BASE_PATH="${{ parameters.service_base_path_pr }}"
+          fi
```

Comment:
Is there a way we could end up with no path here?

Reply:
I.e. not a PR and artifact name doesn't match the pattern

Reply:
The logic to set the service_base_path_pr is in our repo copied over from utils, so this will never be blank but will default incorrectly to "electronic-prescriptions-pr- " if triggered from a branch, so we're not supporting branch deploys. The logic to use this is based on a non-tag build which is all PRs as tags are created on push to master.